com.unity.shadergraph/Editor/Generation/Processors/Generator.cs

```diff
@@ -656,17 +721,12 @@ void GenerateShaderPass(int targetIndex, PassDescriptor pass, ActiveFields activ
             // --------------------------------------------------
             // Dots Instanced Graph Properties
 
-            bool hasDotsProperties = false;
-            m_GraphData.ForeachHLSLProperty(h =>
-            {
-                if (h.declaration == HLSLDeclaration.HybridPerInstance)
-                    hasDotsProperties = true;
-            });
+            bool hasDotsProperties = subShaderProperties.HasDotsProperties();
 
             using (var dotsInstancedPropertyBuilder = new ShaderStringBuilder())
             {
                 if (hasDotsProperties)
-                    dotsInstancedPropertyBuilder.AppendLines(propertyCollector.GetDotsInstancingPropertiesDeclaration(m_Mode));
```

Comment:
It looks like `propertyCollector` still exists here but is effectively useless since it's just collecting things but is never used. If it's too hard to remove, maybe you can just rename it to something like `dummyCollector` with a comment so it's easier to know it's not used anymore.

Reply:
I'll add a comment on it, in the PR going to master:   https://github.com/Unity-Technologies/Graphics/pull/3625